text/0000-let-expression.md

```diff
@@ -0,0 +1,628 @@
+that show us new learners expect it.
+
+This situation is worse with `if let chain` that mix let expressions with `&&` and
+other bools. In fact the compiler will understand it via interpreting let as an
+expression, so why we force humans to understand it another way?
+
+This proposal is also in-line with "everything is an expression" that we have
+in rust.
+
+## Ergonomics
+
+It also available many super powers for us that can
+help decreasing rightward drift without adding to implementation and understanding complexity, and
+actually decreasing it by removing `let-else` and preventing from future similar constructs.
+
+### Compare to `let-else`
+
+```rust
+// simple let else
+let Some(x) = y else {
+    return Err("fail");
+};
+
+// with let expression
+(let Some(x) = y) || {
+    return Err("fail");
+};
+
+// or even better
+(let Some(x) = y) || return Err("fail");
+
+// let else else future possiblity
+let Some(x) = a else b else c else { return; };
+
+// with let expression
+(let Some(x) = a) || (let Some(x) = b) || (let Foo(x) = bar) || { return; };
+```
+
+### New constructs
+
+```rust
+// reuse if let body
+if let Some(x) = a || let Some(x) = b || Ok(x) = c {
+    // body with many lines of code
+} else {
+    // else with many lines of code
+}
+
+// today alternative without code duplication:
+if let ((Some(x), _, _) | (_, Some(x), _) | (_, _, Ok(x)) = (a, b, c) {
+
+// assignment with default
+(let Foo(x) = a) || (let Bar(x) = b) || (let x = default);
+
+// today alternative:
+let x = if let Foo(x) = a { x } else if let Bar(x) = b { x } else { default };
+
+// simple let expression
+assert!((let Some(x) = a) && (let Some(y) = b(x)) && x == y);
+
+// today alternative with if let chains:
+assert!(matches!(a, Some(x) if let Some(y) = b(x) && x == y));
+assert!(if let Some(x) = a && let Some(y) = b(x) && x == y { true } else { false });
+```
+
+## Why now?
+This RFC exists thanks to people who choose `if let` for syntax we know today.
+That syntax wasn't the only choice and there was other options like `iflet`, `if match`, `let if`, `if is` or another `keyword`. 
+If they chose one of the alternatives, no one would have even imagined the let expressions in form of this RFC
+and this RFC either did not come into being at all or came in a completely different
+form and with different capabilities.
+But they chosen `if let` among other options (with good reasons) and we are here. Similarly people
+who choose `&&` for `if let chain` could kill this. They didn't choose `&&` as randomly as
+choosing `if let` and had let expression in their mind, but they had other options like `,` on
+the table, which wasn't compatible with let expressions.
+
+But luck is not always with us. We can't expect each new RFC to randomly add another
+piece of the let expression puzzle to the language. For example `matches!` and `let-else` are
+against this and `let-else` is the possible killer for this RFC. Fortunately, people have
```

Comment:
Your RFC seems to be very against `let-else`, but I can't find any strong arguments in this RFC about why this is better than `let-else`. If your goal is to oppose `let-else` then where is the argument that says *this syntax is superior over `let-else`*?

Reply:
Added: https://github.com/HKalbasi/rfcs/blob/master/text/0000-let-expression.md#let-else-rfc